c++/src/Compression.cc

```diff
@@ -232,6 +239,13 @@ namespace orc {
       }
       rawInputBuffer.resize(0);
     }
+  }
+
+  bool CompressionStream::Next(void** data, int* size) {
+    // triggle compress when rawInputBuffer is reach the capacity
+    if (rawInputBuffer.size() == compressBlockSize) {
```

Comment:
We should make sure `compressionBlockSize` is a multiple of `memoryBlockSize`.

Reply:
I'll add some checks in the writer init function.